packages/shared/tsup.config.ts

```diff
@@ -49,3 +48,28 @@ export const WebWorkerMinifyPlugin: Plugin = {
     });
   },
 };
+
+/**
+ * Preserves import.meta functionality in ESM builds while maintaining ES5 compatibility.
+ * We originally used target: 'es2020' to handle this but it broke support for older browsers (e.g. iOS 12).
+ */
+const preserveImportMetaPlugin: Plugin = {
+  name: 'preserve-import-meta',
+  setup(build) {
+    build.onEnd(result => {
+      if (!result.outputFiles) return;
+
+      result.outputFiles.forEach(file => {
+        if (!file.path.endsWith('.mjs')) return;
+
+        const contents = file.contents;
+        const text = new TextDecoder().decode(contents);
+
+        // Remove esbuild's var import_meta = {} transformation and restore original import.meta references
+        const modified = text.replace(/var\s+import_meta\s*=\s*{};/g, '').replace(/import_meta/g, 'import.meta');
```

Comment:
This restores the original `import.meta` references

Example built file:

<img width="794" alt="Screenshot 2025-02-19 at 9 08 19 PM" src="https://github.com/user-attachments/assets/aec14d4b-8653-42b8-be98-d03bcec7804b" />